r/tests/testthat/test-dataset.R

```diff
@@ -1226,3 +1225,16 @@ test_that("FileSystemFactoryOptions input validation", {
     fixed = TRUE
   )
 })
+
+test_that("can add in augmented fields", {
+  ds <- open_dataset(hive_dir)
+
+  ds <- ds %>%
+    mutate(file_name = add_filename()) %>%
+    collect()
+
+  expect_named(
+    ds,
+    c("int", "dbl", "lgl", "chr", "fct", "ts", "group", "other", "file_name")
+  )
+})
```

Comment:
LGTM, just one question. Would it make sense to add here a test looking into the contents of the `file_name` column? Something along the lines of:
```r
expect_equal(unique(ds$filename), list.files(hive_dir, full.names = TRUE))
```

Reply:
Good call, will add!